addons/website_event_track/models/event_event.py

```diff
@@ -36,11 +36,36 @@ def _compute_website_track(self):
         for event in self:
             if event.event_type_id and event.event_type_id != event._origin.event_type_id:
                 event.website_track = event.event_type_id.website_track
-            elif event.website_menu and (event.website_menu != event._origin.website_menu or not event.website_track):
+            elif event.website_menu and (event.website_menu != event._origin.website_menu):
                 event.website_track = True
             elif not event.website_menu:
                 event.website_track = False
 
+    @api.onchange('website_menu')
+    def _onchange_website_menu(self):
+        """use onchange to make sure that website_track has the correct value
+        when the user makes changes inside the form, what we want is that when
+        the user activates the website menu the website_track field to be set to
+        the value of the template, if the event is not linked
+        to a template, the value of website_track will be set to True.
+        When the menu is deactivated it should always be false"""
+        for event in self:
+            if not event.website_menu:
+                event.website_track = False
+            else:
+                if event.event_type_id:
+                    event.website_track = event.event_type_id.website_track
+                else:
+                    event.website_track = True
```

Comment:
```
if event.website_menu and event.event_type_id:
    event.website_track = event.event_type_id.website_track
else:
    event.website_track = bool(event.website_menu)
```

Reply:
No need for the bool(event.website_menu) though? since website_menu is a boolean.

Reply:
indeed